src/core/frames/frame_controller.ts

```diff
@@ -427,17 +427,43 @@ export class FrameController
   }
 
   private handleFrameMissingFromResponse(fetchResponse: FetchResponse) {
-    this.renderFrameMissingError()
-    this.throwFrameMissingError(fetchResponse)
+    if (this.responsePermittedToEscapeFrame(fetchResponse)) {
+      this.warnResponseEscapingFrame(fetchResponse)
+      this.visitResponse(fetchResponse.response)
+    } else {
+      this.renderFrameMissingError()
+      this.throwFrameMissingError(fetchResponse)
+    }
+  }
+
+  private responsePermittedToEscapeFrame(fetchResponse: FetchResponse) {
```

Comment:
What do you think of `responseShouldEscapeFrame()` to match the "should" convention used elsewhere? I'm thinking of things like `shouldRender`, `shouldCacheSnapshot`, etc.

Reply:
I don't have very strong feelings about this, so am happy to change if you prefer. But at times I find the "should" naming dilutes the intention of methods a bit.

Maybe this one isn't the best example, because "permittedTo" isn't hugely illuminating :) But it does suggest that this is about responses that have been somehow granted the ability to do something, that not every response is allowed to do. I hints at a meaning. Whereas "should" just means "a piece of logic elsewhere will determine this".

I find the "will..." methods are similar sometimes. They are all good conventions to have when crossing API boundaries when it conveys something about the interaction (`shouldX`, `willX`, `didX`) but for internal/private methods I feel like they sometimes get in the way of what could be a clearer name.

It's subtle & subjective for sure, but it can add up I think. What do you think?